### DISCUSSION TITLE
Module parse failed with NextJS

### DISCUSSION THREAD

**Discussion by muradsofi** [TanStack/table#4458] [Q&A]
When I try to use @tanstack/react-table (version 8.5.15) with NextJS (version 10.2.3) I get such an error. I don't understand why this problem happens. Does anyone saw such an issue?  

./node_modules/@tanstack/react-table/build/umd/index.production.js
Module parse failed: Unexpected token (21:1014)
You may need an appropriate loader to handle this file type, currently, no loaders are

**Comment by linhha210498** [comment]
I have the same issue in ReactJS version 17.0.1

**Comment by meezin** [comment]
Any update for this?

**Comment by codereviewvideos** [comment]
this may seem obvious to some, it really wasn't to me.

the problem for me was with my `tsconfig.json`.

I'm not entirely sure which combination of settings was causing this, but afaik I was using the NextJS generated config that would have been created with version 11, I believe. This seems to have worked for ages, but when I came back to the project today and updated all my dependencies, I then hit this error. Strangely, undoing all the changes and trying with the original config also didn't work. 

My workaround / fix was to `npx create-next-app@latest` a new app.

Then literally overwrite my existing `tsconfig.json` with the one generated by the latest nextjs install. 

Now the project boots fine again. 

If this were some enterprise app, I might take the time to comb through the differences and figure out exactly what combination messes it up. But for me, this is just a side project so I'm not going to investigate that further.

Hope it helps.
